federation/pkg/federation-controller/service/service_controller.go

```diff
@@ -614,6 +614,23 @@ func (sc *ServiceController) reconcileService(serviceKey types.NamespacedName) (
 			return statusError, err
 		}
 
+		// Copy ingresses of unreachable clusters for comparison and do not delete those ingresses
```

Comment:
I will try to provide the code as you were suggesting, and i would also test that its possible.
Original code was just not handling the dns deletion upon cluster unregistration